File: cpp/program/Mobil.cpp
#pragma once
#include <bits/stdc++.h> // Mengimpor semua library standar
#include "Kendaraan.cpp"
#include "Mesin.cpp"
#include "Ban.cpp"
#include "Velg.cpp"

using namespace std;

class Mobil : public Kendaraan{
    private:
        string transmisi;
        int jumlahroda;
        Mesin mesin;
        vector<Ban> ban;
        vector<Velg> velg;

    public:
        Mobil():Kendaraan(){
        }

        Mobil(string merk, string nama, string warna, int tahun, string transmisi, int jumlahroda, Mesin mesin, vector<Ban> ban, vector<Velg> velg) : Kendaraan(merk, nama, warna, tahun){
            this->transmisi = transmisi;
            this->jumlahroda = jumlahroda;
            this->mesin = mesin;
            this->ban = ban;
            this->velg = velg;
        }
        
        void set_transmisi(string transmisi){
            this->transmisi = transmisi;
        }

        void set_jumlahroda(int jumlahroda){
            this->jumlahroda = jumlahroda;
        }

        void set_mesin(Mesin mesin){
            this->mesin = mesin;
        }

        void set_ban(vector<Ban> ban){
            this->ban = ban;
        }

        void set_velg(vector<Velg> velg){
            this->velg = velg;
        }

        string get_transmisi(){
            return this->transmisi;
        }

        int get_jumlahroda(){
            return this->jumlahroda;
        }

        Mesin get_mesin(){
            return this->mesin;
        }

        vector<Ban> get_ban(){
            return this->ban;
        }

        vector<Velg> get_velg(){
            return this->velg;
        }

        ~Mobil(){
        }
};

File: python/program/Mobil.py
from Kendaraan import Kendaraan
from Mesin import Mesin
from Ban import Ban
from Velg import Velg

class Mobil(Kendaraan):
    def __init__(self, merk, nama, warna, tahun, transmisi, jumlahroda, mesin, ban, velg):
        super().__init__(merk, nama, warna, tahun)
        self.__transmisi = transmisi
        self.__jumlahroda = jumlahroda
        self.__mesin = mesin if mesin else Mesin()
        self.__ban = ban if ban else []
        self.__velg = velg if velg else []
    
    def set_transmisi(self, transmisi):
        self.__transmisi = transmisi
    
    def set_jumlahroda(self, jumlahroda):
        self.__jumlahroda = jumlahroda
    
    def set_mesin(self, mesin):
        self.__mesin = mesin
    
    def set_ban(self, ban):
        self.__ban = ban
    
    def set_velg(self, velg):
        self.__velg = velg
    
    def get_transmisi(self):
        return self.__transmisi
    
    def get_jumlahroda(self):
        return self.__jumlahroda
    
    def get_mesin(self):
        return self.__mesin
    
    def get_ban(self):
        return self.__ban
    
    def get_velg(self):
        return self.__velg